### ISSUE TITLE
Revenue Management System (RMS) main

### ISSUE THREAD

**Issue by stefanhopman1** [domits1/Domits#436]
## 🚀 Feature Summary

The Revenue Management System (RMS) analyzes historical data, market trends, and booking patterns to optimize pricing and inventory strategies. It helps maximize revenue, improve occupancy, and automate rate adjustments across channels, providing actionable insights for strategic decision-making.

The RMS gets the right guest, the right room, at the right price — but “right” changes by the minute. 
- Reacts in real-time to demand spikes and dips. 
- Optimizes every channel for profitability, not just occupancy. 
- Personalizes offers to guest segments for maximum conversion. 
- Forecasts with precision to outpace the competition. 

----

[PMS](https://github.com/domits1/Domits/issues/149) Capabilities: [Listings](https://github.com/domits1/Domits/issues/2792), [CRS](https://github.com/domits1/Domits/issues/501), [CM](https://github.com/domits1/Domits/issues/440), [Direct Bookings](https://github.com/domits1/Domits/issues/2600), [RMS](https://github.com/domits1/Domits/issues/436), [Calendar & Prices](https://github.com/domits1/Domits/issues/36), [Inbox](https://github.com/domits1/Domits/issues/767), [Tasks](https://github.com/domits1/Domits/issues/2794), [Finance](https://github.com/domits1/Domits/issues/163), [Reviews](https://github.com/domits1/Domits/issues/90), [Intelligence](https://github.com/domits1/Domits/issues/1490), [Settings](https://github.com/domits1/Domits/issues/2793), [Integrations](https://github.com/domits1/Domits/issues/433), [Agentic AI](https://github.com/domits1/Domits/issues/442)
Use Case / Scenario Testing: [PMS](https://github.com/domits1/Domits/issues/2262), [CRS](https://github.com/domits1/Domits/issues/2265), [CM](https://github.com/domits1/Domits/issues/2266), [RMS](https://github.com/domits1/Domits/issues/2267), [Calendar & Prices](https://github.com/domits1/Domits/issues/2299), [Inbox](https://github.com/domits1/Domits/issues/2291), [Finance](https://github.com/domits1/Domits/issues/2287)

----

<img width="1306" height="517" alt="Image" src="https://github.com/user-attachments/assets/18831730-9e64-428b-8897-7acc438ba687" />

----
**Reporting & Analytics**
- Understand [55 Key Performance Indicators (KPI’s) for Hospitality](https://bookdomits.com/key-performance-indicators-kpis-hospitality/) 
- Understand: [Advanced Dynamic Pricing & Revenue Management For Hospitality](https://bookdomits.com/advanced-dynamic-pricing-revenue-management-hospitality/)
- [x] #2334
- [ ] #2335 

**Rate Management**
- [ ] #2393
- [ ] #1501 
- [ ] Advanced Fees Setup

**Integrations**
- [x] #2297 
- [ ] #2927 

---

## 🧩 Problem Statement
What problem does this feature solve? Who is affected, and why does it matter? > Focus on the problem, not the solution.

No Centralized Revenue Management System (RMS)

---

## 🎯 Desired Outcome
What does success look like if this feature is implemented?

Centralized Revenue Management System (RMS)

---

## 👥 Target Users
Who benefits from this feature?
- [ ] Guests / Customers
- [x] Hosts / Property Managers / Enterprise
- [ ] Admin / Internal users
- [ ] Developers
- [ ] Other: _______

---

## 📈 Business & User Impact
How valuable is this feature?

- **Impact level:**
  - [ ] Low (nice-to-have)
  - [ ] Medium (improves workflows)
  - [ ] High (unblocks key use cases)
  - [x] Critical (strategic / revenue-impacting)

- **User reach:**
  - [ ] Few users
  - [ ] Many users
  - [x] All users
  - [ ] Unknown

---

## ⚠️ Constraints & Considerations
- Performance
- Security
- Privacy
- Accessibility
- Backward compatibility
- External dependencies

---

## 🧪 Acceptance Criteria / User Stories
How can we validate that this feature is complete?

- [ ] Given X, when Y, then Z

---

## Related Issues / PRs
Link any related discussions, issues, or pull requests.

- [ ] Context: [55 Key Performance Indicators (KPI’s) for Hospitality](https://bookdomits.com/key-performance-indicators-kpis-hospitality/) 
- [ ] Context: [Advanced Dynamic Pricing & Revenue Management For Hospitality](https://bookdomits.com/advanced-dynamic-pricing-revenue-management-hospitality/)
- [ ] https://github.com/domits1/Domits/issues/510

----

Extra research below:

**Second phase**

**Visualization tools**
- [x] Graphs/Charts: Time-series trends for revenue, occupancy, ADR, etc.
- [x] Heatmaps: For booking windows or occupancy by day of the week.
- [x] KPIs At-a-Glance: Summarize top metrics like RevPAR, ADR, and occupancy on a single screen.
- [ ] Filters: Segment by location, market, or guest type.

----

- [x] Stripe revenue Recognition: https://stripe.com/en-nl/revenue-recognition
- [ ] Automated revenue reports and dashboards
- [ ] Views
- [ ] Clicks
- [ ] Look-to-Book (L2B) Ratio
- [x] Nights available
- [x] Nights booked
- [ ] Booking window
- [ ] Number of check-ins
- [x] Occupancy rates
- [ ] Target occupancy rates
- [ ] Fixed flat rate
- [ ] Minimum Accepted Rate (MAR)
- [x] ADR (Average Daily Rate) [Total Room Rev / Rooms Sold]
- [x] Target ADR (Average Daily Rate) 
- [x] Gross revenue
- [x] Target gross revenue
- [ ] Room Revenue
- [x] ALOS (Average Length of Stay)
- [x] RevPAR (Revenue per Available Room) [Total Room Revenue / Rooms Sold]
- [ ] Target RevPAR 
- [ ] Bookingtrends
- [ ] NOI (Net Operating Income). (Gross revenue - gross operating expenses)
- [ ] Target NOI
- [ ] Earning Potential
- [x] Revenue Growth Rate (YoY or MoM)
- 
- Channel Revenue Contribution
- [ ] Bookings by channel (Direct Bookings, OTA (Online Travel Agencies), GDS (Global Distribution Systems))
- [ ] Revenue by channel (Direct Bookings, OTA (Online Travel Agencies), GDS (Global Distribution Systems))
- 
- [ ] TRevPAR (Total Revenue per Available Room)
- [ ] Target TRevPAR
- [ ] RevPAC (Revenue per Available Customer)
- [ ] Target RevPAC
- [ ] GOPPAR (Gross Operating Profit per Available Room)
- [ ] Target GOPPAR
- [ ] ARPAR or Adjusted RevPAR (Adjusted Revenue per Available Room)
-
- Operational cost metrics
- [ ] - [ ] Management fees (10%, 15%, 20%, 25%, 30%, custom%)
- [ ] Target management fees
- [ ] Operational expenses (10%, 15%, 20%, 25%, 30%, custom%)
- [ ] Target operational expenses
- [ ] CPOR (Cost per Occupied Room)
- [ ] Target CPOR
- [ ] Labor costs as a % of revenue
- [ ] Housekeeping efficiency
- [ ] Energy consumption per room [sustainability metric]
-
- Revenue Segmentation
- [ ] Guest type: Business, leisure, group, transient
- [ ] Region: Domestic, international
- [ ] Age demographic, other pscyhographics
-
- Market and competitive metrics
- [ ] Market Penetration Index (MPI) [Compares occupancy rates]
- [ ] Average Rate Index (ARI) [Compares ADR]
- [ ] Revenue Generation Index (RGI) [Compares RevPAR
- [ ] Comp Set Analysis [Benchmarking performance in specific market]
-
- Digital Marketing Metrics
- [ ] Conversion Rate
- [ ] Cost Per Acquisition (CPA)
- [ ] Click-Through Rate (CTR)
- [ ] Traffic Sources
- [ ] Social Media Engagement
- [ ] Search Engine
- [ ] Online Booking Abandonment Rate
-
- Guest Experience and Satisfaciton metrics
- [ ] Net Promoter Score (NPS) [Guest loyalty and recommendation]
- [ ] Guest Satisfaction Scores (GSS)
- [ ] Average Review Rating
- [ ] Service Recovery Rate
- [ ] Check-in and check-out times
- [ ] Guest retention rate
-
- Forecasting
- [ ] Revenue Forecast
- [ ] Occupancy Forecast
- [ ] ADR Forecast
- [ ] RevPar Forecast
- [ ] NOI Forecast
-
- Break-Even
- [ ] Revenue Break-Even
- [ ] Occupancy Break-Even
- [ ] ADR Break-Even
- [ ] RevPar Break-Even
- [ ] NOI Break-Even
-
- Sustainability Metrics
- [ ] Carbon Footprint Per Room 
- [ ] Waste Reduction Rate
- [ ] Water and Energy Efficiency

**Third phase**
- [ ] #1339
- [ ] #435

**Later**
- [ ] ROI Score
- [ ] Earnings Before Interest, Tax, Depreciation and Amortisation (EBITDA)
- [ ] Internal Rate of Return (IRR)
- [ ] Net Present Value (NPV)
- [ ] Payback Period
- [ ] Home Value (HV)
- [ ] Annual Revenue (AR)
- [ ] AR to HV ratio
- [ ] # Active properties
- [ ] Tax

Examples to study
-  [MyLightHouse](https://www.mylighthouse.com/)
- [Pricelabs](https://hello.pricelabs.co/)
-  [Wheelhouse](https://www.usewheelhouse.com/navigator)
-  [Beyondpricing](https://www.beyondpricing.com/)
-  [Rategain](https://rategain.com/)
-  [AirDNA](https://www.airdna.co/)
-  [Duetto](https://www.duettocloud.com/en-us/)
- [Ideas](https://ideas.com/)
- [LodgiQ](https://lodgiq.com/)
- [RoomPriceGenie](https://roompricegenie.com/nl/)
- [Atomize](https://atomize.com/)
- [Flyr](https://flyr.com/)
-  STR

Resources
- [ ] https://www.altexsoft.com/blog/vacation-rental-revenue-management/
- [ ] https://www.altexsoft.com/blog/revpar-occupancy-rate-adr-hotel-metrics/
- [ ] https://www.altexsoft.com/blog/hotel-revenue-management-solutions-best-practices-revenue-managers-role/
- [ ] https://www.mews.com/en/blog/what-is-revenue-management
- [ ] https://www.cloudbeds.com/revenue-management/
- [ ] https://skift.com/skift-short-term-rental-250/

Later we can add a revenue tool so the host can see in the onboarding his earning potential. 
- [ ] https://github.com/domits1/Domits/issues/190
- [ ] https://github.com/domits1/Domits/issues/239

---
Resource property metrics: 
https://www.altexsoft.com/blog/revpar-occupancy-rate-adr-hotel-metrics/

**Occupancy Rate**
Occupancy rate is measured by dividing the number of occupied rooms by the number of available rooms and multiplying by 100, showing the percentage of rooms occupied at a specific moment.
For example, if you have a 10-room hotel and last night you sold 5 rooms, then the occupancy rate would be 50 percent. In further calculations we will use 0.5 which is the same as 50 percent.

**ADR (Average Daily Rate)**
ADR is used to calculate the average rental revenue per occupied room at a given time. To find ADR, divide your total room revenue by the number of rooms sold.
For example, if you sold 5 rooms out of your 10-room hotel and your total revenue was $2,000, then ADR would be $400. Obviously, it makes sense to calculate the metric only if you have different room rates, otherwise its value is pretty evident.

**ALOS (Average Length of Stay)**
ALOS identifies the average number of nights your guests spend at your hotel and is calculated by dividing the total room nights by the total number of bookings.
For example, in April the number of room nights was 200 (out of maximum 300 for a 10-room hotel), with a total of 50 bookings. Then, ALOS is 4 days.

---

**RevPAR (Revenue per Available Room)**
RevPAR shows the amount of revenue generated by one room, whether booked or not. There are two ways to calculate it. You can either divide your total room revenue by the total number of available rooms OR multiply ADR by the occupancy rate.
For example, selling 5 rooms out of 10 brought you $2,000, so your RevPAR equals $200 (you’re getting the same result by multiplying your ADR of $400 by the occupancy rate of 0.5.)

**TRevPAR (Total Revenue per Available Room)**
TRevPAR is measured by dividing your total revenue (from room rental and other sources) by the total number of available rooms. This metric shows the amount of total revenue of the property per one room (if rented out seperately.
For example, you have a revenue of $2,000 from the 5 booked rooms out of ten available rooms. The revenue of the restaurant is $1,000 and the revenue of the spa is $500. With the total gross revenue of $3,500, TRevPAR equals $350.

**RevPAC (Revenue per Available Customer)**
RevPAC shows the amount of revenue per guest. To define it, divide the total revenue by the total number of property guests. You can also make the calculation including customers that do not stay at the property, but only come to use the amenities, if you have such an option for visitors.
For example, you sold 3 double rooms and 2 family rooms for 4 people which brought you $2000. They ate at the restaurant for a total of $1,000 and used the spa services for $500. Besides, 16 people visited your bar, resulting in another $700. Then, your RevPAC per each of your 30 guests would come to $140.

**GOPPAR (Gross Operating Profit per Available Room)**
GOPPAR demonstrates how much gross operating profit comes from each room. Gross operating profit is your total revenue minus gross operating expenses, and if you divide the result by the total number of available rooms, you’ll obtain your GOPPAR.
For example, your total net revenue is $3,500 and your gross expenses are $1,700. So gross operating profit would be $1800. Divided by the 10 rooms, GOPPAR equals $180. If you want to see the yearly GOPPAR, you have to use the yearly values of the revenue and expenses and multiply the number of rooms by 365.

**CPOR (Cost per Occupied Room)**
CPOR is measured by simply dividing gross operating expenses by the number of rooms sold to show how much every sold room costs you.
For example, you have 5 rooms occupied and your total operating expenses are $1,700. The CPOR would be $340.

**ARPAR or Adjusted RevPAR (Adjusted Revenue per Available Room)**
ARPAR shows how much profit each room brings you, considering variable costs (we’ve explained this figure in the GOPPAR section) and additional revenue. To calculate ARPAR, first subtract the variable costs per sold room from your ADR, add the amount of additional revenue per sold room, and then multiply the result by your occupancy rate.
For example, your ADR is $400 and your occupancy is 50% with 5 rooms sold out of 10. Let’s say your variable costs are $750 (which is $150 per room) and the additional revenue is $1,500 (which is $300 per room). Then, your ARPAR would be $275.

**NOI (Net Operating Income)**
An NOI shows whether your property is making a profit or loss. To find it you have to subtract your gross operating expenses from your total revenue.
For example, you made $3,500 from room sales, restaurant, and spa. Your gross expenses are $1,700, making that day's NOI $1,800. To calculate your weekly, monthly, or yearly NOI, you have to take the corresponding total figures.

**Comment by Timmhart** [comment]
## User story
As a host, I want to view my revenue from the last month on my dashboard, so I can track my earnings and make informed decisions for future bookings.

## Example mapping

- What if the host has no revenue in the past month?
   - Display a message like: "No revenue data available for the selected period." Provide empty visuals (charts/graphs) and offer options to view previous months or highlight future booking opportunities.
- What if the host manages multiple properties?
   - Provide a filter or tab to view revenue by individual property, with a cumulative view for overall revenue across all properties.
- What if there is a system issue retrieving the data?
   - Display an error message indicating that revenue data is temporarily unavailable, and provide an option to refresh or contact support.
- What if the revenue data doesn’t align with the host’s expectations?
   - Include a breakdown of deductions (e.g., service fees, taxes) and provide an option for the host to report discrepancies for review.
- What if the host wants to view revenue beyond just the last month?
  - Allow the host to choose other time periods via a date picker (e.g., "Last 3 months" or "Year to date") to see more historical data.
- What if the host prefers different visualization formats?
  - Offer options to switch between visualizations like bar graphs, line graphs, and pie charts for better revenue trend analysis.

## Accept criteria

- [ ] The dashboard should display the total revenue for the past month, along with a comparison to the previous month.
- [ ] A breakdown of revenue by property (if multiple) should be available.
- [ ] The dashboard should show key metrics such as average daily rate, occupancy rate, and number of bookings.
- [ ] The revenue data should be downloadable as a CSV or PDF report.
- [ ] A visual graph or chart should illustrate the revenue trends for the month.
- [ ] If no revenue data is available for the selected period, the dashboard should display empty visuals (graphs/charts) with a message like "No revenue data available for the selected period."
- [ ] The empty visuals should provide an option to view revenue from previous months or highlight upcoming booking opportunities.
